pkg/pillar/cmd/zedkube/applogs.go

```diff
@@ -176,10 +180,39 @@ func (z *zedkube) checkAppsStatus() {
 		}
 		log.Functionf("checkAppsStatus: devname %s, pod (%d) status %+v, old %+v", z.nodeName, len(pods.Items), encAppStatus, oldStatus)
 
+		// Publish if there is a status change
 		if oldStatus == nil || oldStatus.IsDNidNode != encAppStatus.IsDNidNode ||
 			oldStatus.ScheduledOnThisNode != encAppStatus.ScheduledOnThisNode || oldStatus.StatusRunning != encAppStatus.StatusRunning {
 			log.Functionf("checkAppsStatus: status differ, publish")
 			z.pubENClusterAppStatus.Publish(aiconfig.Key(), encAppStatus)
 		}
 	}
 }
+
+func (z *zedkube) handleKubePodsGetError(items, stItems map[string]interface{}) {
+	if z.getKubePodsError.getKubePodsErrorTime == nil {
+		now := time.Now()
+		z.getKubePodsError.getKubePodsErrorTime = &now
+		log.Noticef("handleKubePodsGetError: can't get pods, set error time")
+	} else if time.Since(*z.getKubePodsError.getKubePodsErrorTime) > 2*time.Minute {
+		if z.getKubePodsError.processedErrorCondition == false {
+			z.getKubePodsError.processedErrorCondition = true
+			for _, item := range items {
+				aiconfig := item.(types.AppInstanceConfig)
+				for _, st := range stItems {
+					aiStatus := st.(types.ENClusterAppStatus)
+					if aiStatus.AppUUID == aiconfig.UUIDandVersion.UUID {
+						// if we used to publish the status, of this app is scheduled on this node
+						// need to reset this, since we have lost the connection to the kubernetes
+						// for longer time than the app is to be migrated to other node
```

Comment:
Are you assuming that time is the hard-coded 2 minutes above?

Reply:
Yes. The logic here is when checking the app-status through k3s, if hit error, we wait until if the error persist for more than 2 minutes. if yes, the code walks through all the app, if used to set 'scheduled on this node', we reset this.

Reply:
But do we know that k3s has failed after 2 minutes? Or could it take longer?

Reply:
We do not know for sure. But the setting we have for the k3s is that, after 1 minute the node will be declared by the cluster as 'not ready', and after 30 minutes, the app suppose to be 'migrated' to another 'ready' node. So, in one and half minutes, the app suppose to be gone. although this is from cluster point of view. the code here is from a node point of view and we can not talk to our  node, and assumption is that, our cluster neighbors also can not talk to our node and the 'migration' will happen within 2 minutes.

Reply:
I'll add some comments in this area.